contrib/docs/tutorials/aws-alb-aad-oidc-auth.md

```diff
@@ -0,0 +1,175 @@
+- Select `OIDC` under `Authenticate`
+- Set `Issuer` to `https://login.microsoftonline.com/{TENANT_ID}/v2.0`, replacing `{TENANT_ID}` with the value of `Directory (tenant) ID` of the AAD App.
+- Set `Authorization endpoint` to `https://login.microsoftonline.com/{TENANT_ID}/oauth2/v2.0/authorize`, replacing `{TENANT_ID}` with the value of `Directory (tenant) ID` of the AAD App.
+- Set `Token endpoint` to `https://login.microsoftonline.com/{TENANT_ID}/oauth2/v2.0/authorize`, replacing `{TENANT_ID}` with the value of `Directory (tenant) ID` of the AAD App.
+- Set `User info endpoint` to `https://graph.microsoft.com/oidc/userinfo`
+- Set `Client ID` to the AAD App `Application (client) Id`
+- Set `Client secret` to the AAD APP `client secret`
+
+Use the following advanced settings:
+
+- `Session cookie name` = `AWSELBAuthSessionCookie`
+- `Session timeout` = `604800` seconds
+- `Scope` = `openid profile offline_access`
+- `Action on unauthenticated request` = `Autenticate (client reattempt)`
+
+Once you've saved the action, you should see an authentication flow be triggered against AAD when visiting Backstage address at `https://backstage.yourdomain.com`. The flow will not complete successfully as Backstage app isn't yet configured properly.
+
+## Backstage changes
+
+### Frontend
+
+The Backstage App needs a SingInPage when authentication is required.
+When using ALB authentication Backstage will only be loaded once the user has successfully authenticated; we won't need to display a SignIn page, however we will need to create a dummy SignIn component that can refresh the token.
+
+- edit `packages/app/src/App.tsx`
+- import the following two additional definitions from `@backstage/core`: `useApi`, `configApiRef`; these will be used to check wether backstage is running locally or behind an ALB
+- add the following definition just before the app is created (`const app = createApp`):
+
+```ts
+const DummySignInComponent: any = (props: any) => {
+  const config = useApi(configApiRef);
+  const shouldAuth = !!config.getOptionalConfig('auth.providers.awsalb');
+  if (shouldAuth) {
+    fetch(`${window.location.origin}/api/auth/awsalb/refresh`)
+      .then(data => data.json())
+      .then(data => {
+        props.onResult({
+          userId: data.backstageIdentity.id,
+          profile: data.profile,
+        });
+      });
+  } else {
+    // when running locally we default user identity to `Local User`
+    props.onResult({
+      userId: 'local user',
+      profile: {
+        email: 'local.user@yourdomain.com',
+        displayName: 'Local User',
+        picture: '',
+      },
+    });
+  }
+
+  return <div />;
+};
+```
+
+- use `DummySingInComponent` as `SignInPage` by changing `createApp` statement as follow:
+
+```ts
+const app = createApp({
+  apis,
+  plugins: Object.values(plugins),
+  components: {
+    SignInPage: DummySignInComponent,
+  },
+  bindRoutes({ bind }) {
```

Comment:
I'd say omit this and just highlight the addition to `components` instead. Something like 

```ts
const app = createApp({
  ...
  components: {
    SignInPage: DummySignInComponent,
    ...
  },
  ...
})
```